cpp/include/cudf/wrappers/timestamps.hpp

```diff
@@ -32,23 +32,26 @@ namespace cudf {
 namespace detail {
 // TODO: Use chrono::utc_clock when available in libcu++?
 template <class Duration>
-using time_point = simt::std::chrono::time_point<simt::std::chrono::system_clock, Duration>;
+using time_point = simt::std::chrono::sys_time<Duration>;
 
 template <class Duration>
 struct timestamp : time_point<Duration> {
+  // Bring over base class constructors and make them visible here
+  using time_point<Duration>::time_point;
+
+  // This is needed as __shared__ objects of this type can't be assigned in device code
```

Comment:
I believe this is only a problem with `simt::std::chrono` and not an issue with `cuda::std::chrono`, yes?

Reply:
i *think* this also shows up with `cuda::std::chrono` as well. assignments of timestamps to the `__shared__` device variable will trigger this